src/wp-includes/media.php

```diff
@@ -3848,6 +3848,14 @@ function wp_get_image_editor( $path, $args = array() ) {
 		}
 	}
 
+	$mime_type = isset( $args['mime_type'] ) ? $args['mime_type'] : wp_check_filetype( $path )['type'];
```

Comment:
Why is this logic separated from the logic above re: $args['mime_type']?

Is it possible to combine them? If not, could you add a comment explaining what is happening in this line?

Reply:
Good question! I clarified the block above with a doc block and changed this conditional to set the mime type to empty if `$args['mime_type']`  isn't set - it already tries extracting the mime type from the file above.